pyked/chemked.py

```diff
@@ -608,12 +610,20 @@ def __init__(self, properties):
                 setattr(self, prop.replace('-', '_'), None)
 
         self.composition_type = properties['composition']['kind']
-        composition = deepcopy(properties['composition']['species'])
-
+        # composition = deepcopy(properties['composition']['species'])
+        composition = []
+        for spec in properties['composition']['species']:
+            species_name = spec['species-name']
+            amount = spec['amount']
+            InChI = spec.get('InChI')
+            SMILES = spec.get('SMILES')
+            atomic_composition = spec.get('atomic-composition')
+            composition.append(Composition(species_name=species_name, InChI=InChI, SMILES=SMILES, atomic_composition=atomic_composition, amount=amount))
+        print(composition)
         for idx, species in enumerate(composition):
```

Comment:
You can combine this loop with the previous loop by doing the processing on `spec['amount']` when you get the value out of the dictionary. That will be more efficient. You can probably just delete this line, and move your line 621 (`composition.append(...)`) to line 659. You'll also have to change lines 624->626 to use `spec['amount']` rather than `species['amount']` but everything else should stay the same